### DISCUSSION TITLE
Not an issue - Question about translations and pull request process

### DISCUSSION THREAD

**Discussion by LagoonProject** [romcal/romcal#430] [Q&A] [answered]
Hi,

I would like to try to provide some translations for the locales in Spanish and Portuguese, as I see there's quite some days that the name entry falls back in english.

What are the files that should be updated? I see in the locales folder there is es.ts and pt-br.ts, Is there something else to update?

My idea is to take some keys in general-calendar > proper-of-saints.ts and also in some particular-calendars that are not in es.ts and pt-br.ts and see if I can provide the translation. Is that the right thing to do?

Should I paste the new entries below the existing ones so it's easier for review or should I respect the alphabetical order?

Is there anything else to do for the pull request of translations?

Thanks.

**Comment by emagnier** [comment, marked as answer]
Hello @LagoonProject! 

Thank you for offering your help on the Spanish and Portuguese translations, you are more than welcome to help us on this project!

Romcal 3.x ins't released yet, and there is still unfortunately a lack of documentation, contribution tutorials, and tooling to provide easy ways to bootstrap & check locales and calendar data... 

But indeed, as you see, there are already some locale files that could be improved/completed:
- `lib/locales/es.ts`
- `lib/locales/pt-br.ts`
If you are only working on translation, this should be the only 2 files you should have to edit.

FYI, the English locale (`en.ts`) should contain everything localized. So if you see that something is missing, you can have a look at the English locale file to have an example.

And it's better to keep the alphabetical order. Reviewing them in GitHub will not be a problem.

Thank again!

**Reply by tukusejssirs** [reply]
@LagoonProject, all @emagnier said is true, I just want to add additional information.

There are three parts of contributing to romcal, and two additional:
- [martyrology] if you need to add a saint/blessed/celebration not yet in romcal codebase, you need to add it to `lib/catalog/martyrology.ts`;
- [calendars] if you want to add celebrations to a particular calendar, you need to define it in `lib/particular-calendars/*.ts` (`lib/particular-calendars/index.ts` re-exports all particular calendars);
- [locales] if you want to add a localised celebration name (or anything that is defined in `lib/locales/en.ts`) in a particular locale, you need to define it in `lib/locales/*.ts`;
- [docs] if you want to help us with the documentation, you can create it in Markdown under `docs/`.

Anyway, same question was already asked in #402 and #378, so I suggest you should read all those comments in #402 (you could also check out #379, #403). All in all, I would like to ask you to provide sources for any addition you add (e.g. list of celebrations inscribed in a calendar, celebration name localisation), although that is not a hard requirement, but I would like to improve this in order to make romcal data verifiable.

Also, I think that creating a WIP/draft PR as soon as possible is a good way to make sure your work will be reviewed ASAP, while you will not need to refactor/re-work much along the process.

**Comment by LagoonProject** [comment]
@emagnier @tukusejssirs  Thanks for your detailed answers, I'll make sure to comply as much as possible to the requirements.

**Reply by tukusejssirs** [reply]
We’ll guide you, don’t worry. :wink:
